pidgin/app.py

```diff
@@ -57,57 +62,60 @@ def flatten_dict(d):
                     flat_d[k] = v
             else:
                 flat_d[k] = v
-    except Exception:
+    except (AttributeError, IndexError):
         error = 'Core metadata not available for this file'
         if 'errors' in d:
             error += ': ' + d['errors'][0]
-        raise Exception(error)
+        raise NoCoreMetadataException(error)
     return flat_d
 
 
-# Get the type of file from the object_id.
 def get_file_type(object_id):
+    """
+    Get the type of file from the object_id.
+    """
     query_txt = '{ datanode (object_id: "' + object_id + '") { type } }'
     response = send_query(query_txt)
     try:
         file_type = response.get_json()['data']['datanode'][0]['type']
     except IndexError:
-        raise Exception('object_id "' + object_id + '" not found')
+        raise ObjectNotFoundException('object_id "' + object_id + '" not found')
     return file_type
 
 
-# Write a query and transmit it to send_query().
 def request_metadata(object_id):
-    # get the metadata from the type of file and the object_id
+    """
+    Write a query and transmit it to send_query().
+    """
     file_type = get_file_type(object_id)
-    query_txt = '{ ' + file_type + ' (object_id: "' + object_id + """") {
-        core_metadata_collections {
+
+    # get the metadata from the type of file and the object_id
+    query_txt = '''{{ {} (object_id: "{}") {{
+        core_metadata_collections {{
             title description creator contributor coverage
             language publisher rights source subject
-        }
+        }}
         type file_name data_format file_size
-        project_id object_id updated_datetime }
-    }"""
+        project_id object_id updated_datetime }} }}
+        '''.format(file_type, object_id)
```

Comment:
oh no, I forgot about this `{{` thing in format strings...I guess I'll leave it up to you whether you prefer this or the string concatenation